odxtools/cli/main.py

```diff
@@ -50,4 +60,12 @@ def start_cli() -> None:
 
     for tool in tool_modules:
         if tool._odxtools_tool_name_ == args.subparser_name:
-            tool.run(args)
+            orig_strict = odxtools.exceptions.strict_mode
+            odxtools.exceptions.strict_mode = not args.no_strict
+            try:
+                tool.run(args)
+            except:
+                raise
```

Comment:
try/finally enough, the except block is not required here

Reply:
that's true if no backtrace ought to be printed if an exception happens to seep out of the tool's `run()` function and if `start_cli()` is guaranteed to never be called from other python code. I think that of these two assertions,  only the latter applies. (I would prefer to be have a fighting chance of finding bugs based on the output of the CLI tools...)

Reply:
(that's due to the fact that so far, no non-standalone callers `start_cli()` exist: `test_cli.py` calls the `run()` functions of the tools directly.)

Reply:
I meant the code should look like this

```python
            try:
                tool.run(args)
            finally:
                odxtools.exceptions.strict_mode = orig_strict
```

Reply:
okay, my mistake: I somehow assumed that this way exceptions thrown by `tool.run()` get silently ignored whilst in reality they get passed through. (I don't know what made me think that, probably a brain fart :/. anyway: fixed.)